internal/traceutil/init.go

```diff
@@ -31,6 +32,13 @@ func InitTraceExporter(ctx context.Context, serviceName string, batchOptions ...
 
 	logger := slog.With("component", "trace")
 
+	// OTEL_SDK_DISABLED is not supported by the Go SDK, but is a standard env
+	// var defined by the OTel spec. We'll use it to disable the trace provider.
+	if disabled, _ := strconv.ParseBool(os.Getenv("OTEL_SDK_DISABLED")); disabled {
+		logger.Debug("Tracing disabled by environment variable")
```

Comment:
nit: suggest adding which env var trigged the disable into the log line? Might just make it more obvious

Reply:
ah good point. done!